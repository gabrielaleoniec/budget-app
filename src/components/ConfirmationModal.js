import React from 'react';

const ConfirmationModal = (props) => {
    const className = props.showConfirmationBox ? "confirmation-box--show" : '';
    return <div className="box-layout box-layout--cover">
        <div className={`confirmation-box ${className}`}>
            <p>Should the expense be removed?</p>
            <div className="confirmation-box__buttons">
                <button className="button button--weak" onClick={props.hideConfirmationModal}>Cancel</button>
                <button className="button button--remove" onClick={props.removeExpense}>Remove</button>
            </div>
        </div>
    </div>
}

export default ConfirmationModal;
