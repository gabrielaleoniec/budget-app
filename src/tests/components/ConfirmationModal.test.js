import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationModal from '../../components/ConfirmationModal';

let history, removeExpense, hideConfirmationModal, showConfirmationBox, wrapper;
beforeEach(() => {
    removeExpense = jest.fn();
    hideConfirmationModal = jest.fn();
    showConfirmationBox = false;
    history = { push: jest.fn() };
    wrapper = shallow(<ConfirmationModal
        removeExpense={removeExpense}
        hideConfirmationModal={hideConfirmationModal}
        showConfirmationBox={showConfirmationBox} />);
});

test('Should render ConfirmationModal correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.confirmation-box--show').length).toBe(0);
});

test('Should have proper css class on showConfirmationBox true', () => {
    wrapper = shallow(<ConfirmationModal
        removeExpense={removeExpense}
        hideConfirmationModal={hideConfirmationModal}
        showConfirmationBox={true} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.confirmation-box--show')).toHaveLength(1);
});

test('Should call startRemoveExpense with right properties on button click ', () => {
    wrapper.find('button').last().simulate('click');
    expect(removeExpense).toHaveBeenCalled();
});

test('Should call startRemoveExpense with right properties on button click ', () => {
    wrapper.find('button').first().simulate('click');
    expect(hideConfirmationModal).toHaveBeenCalled();
});