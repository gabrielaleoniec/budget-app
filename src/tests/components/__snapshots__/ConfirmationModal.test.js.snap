// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have proper css class on showConfirmationBox true 1`] = `
<div
  className="box-layout box-layout--cover"
>
  <div
    className="confirmation-box confirmation-box--show"
  >
    <p>
      Should the expense be removed?
    </p>
    <div
      className="confirmation-box__buttons"
    >
      <button
        className="button button--weak"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        className="button button--remove"
        onClick={[MockFunction]}
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;

exports[`Should render ConfirmationModal correctly 1`] = `
<div
  className="box-layout box-layout--cover"
>
  <div
    className="confirmation-box "
  >
    <p>
      Should the expense be removed?
    </p>
    <div
      className="confirmation-box__buttons"
    >
      <button
        className="button button--weak"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        className="button button--remove"
        onClick={[MockFunction]}
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
